.github/workflows/ci.yml

```diff
@@ -0,0 +1,83 @@
+name: CI tests
+
+on: [push, workflow_dispatch]
+
+jobs:
+  linux:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake_opts:
+          - '-DCMARK_SHARED=ON'
+          - ''
+        compiler:
+          - c: 'clang'
+            cpp: 'clang++'
+          - c: 'gcc'
+            cpp: 'g++'
+    env:
+       CMAKE_OPTIONS: ${{ matrix.cmake_opts }}
+       CC: ${{ matrix.compiler.c }}
+       CXX: ${{ matrix.compiler.cpp }}
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install valgrind
+      run: |
+         sudo apt install -y valgrind
+    - name: Build and test
+      run: |
+         make
+         make test
+         make leakcheck
+
+  macos:
+
+    runs-on: macOS-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake_opts:
+          - '-DCMARK_SHARED=ON'
+          - ''
+        compiler:
+          - c: 'clang'
+            cpp: 'clang++'
+          - c: 'gcc'
```

Comment:
does that make sense on macOS

Reply:
If gcc is available on macOS in GitHub Actions (i assume so since this is an upstream change), i guess they want to make sure to test it. However, this doesn't affect our fork, since we don't use GitHub Actions.